jenkins/databricks/build.sh

```diff
@@ -15,437 +15,303 @@
-   -DartifactId=arrow-format \
-   -Dversion=$SPARK_VERSION_TO_INSTALL_DATABRICKS_JARS \
-   -Dpackaging=jar
-
-$MVN_CMD -B install:install-file \
-   -Dmaven.repo.local=$M2DIR \
-   -Dfile=$JARDIR/$ARROWMEMORYJAR \
-   -DgroupId=org.apache.arrow \
-   -DartifactId=arrow-memory \
-   -Dversion=$SPARK_VERSION_TO_INSTALL_DATABRICKS_JARS \
-   -Dpackaging=jar
-
-$MVN_CMD -B install:install-file \
-   -Dmaven.repo.local=$M2DIR \
-   -Dfile=$JARDIR/$ARROWVECTORJAR \
-   -DgroupId=org.apache.arrow \
-   -DartifactId=arrow-vector \
-   -Dversion=$SPARK_VERSION_TO_INSTALL_DATABRICKS_JARS \
-   -Dpackaging=jar
-
-$MVN_CMD -B install:install-file \
-   -Dmaven.repo.local=$M2DIR \
-   -Dfile=$JARDIR/$ORC_CORE_JAR \
-   -DgroupId=org.apache.orc \
-   -DartifactId=orc-core \
-   -Dversion=$SPARK_VERSION_TO_INSTALL_DATABRICKS_JARS \
-   -Dpackaging=jar
-
-$MVN_CMD -B install:install-file \
-   -Dmaven.repo.local=$M2DIR \
-   -Dfile=$JARDIR/$ORC_SHIM_JAR \
-   -DgroupId=org.apache.orc \
-   -DartifactId=orc-shims \
-   -Dversion=$SPARK_VERSION_TO_INSTALL_DATABRICKS_JARS \
-   -Dpackaging=jar
-
-$MVN_CMD -B install:install-file \
-   -Dmaven.repo.local=$M2DIR \
-   -Dfile=$JARDIR/$ORC_MAPREDUCE_JAR \
-   -DgroupId=org.apache.orc \
-   -DartifactId=orc-mapreduce \
-   -Dversion=$SPARK_VERSION_TO_INSTALL_DATABRICKS_JARS \
-   -Dpackaging=jar
-
-$MVN_CMD -B install:install-file \
-   -Dmaven.repo.local=$M2DIR \
-   -Dfile=$JARDIR/$PROTOBUF_JAR \
-   -DgroupId=com.google.protobuf \
-   -DartifactId=protobuf-java \
-   -Dversion=$SPARK_VERSION_TO_INSTALL_DATABRICKS_JARS \
-   -Dpackaging=jar
+# This script installs dependencies required to build RAPIDS Accelerator for Apache Spark on DB.
+# Usage: `BASE_SPARK_VERSION=3.2.1 ./jenkins/databricks/build.sh
+# Environments SPARKSRCTGZ, BASE_SPARK_VERSION, BASE_SPARK_VERSION_TO_INSTALL_DATABRICKS_JARS, MVN_OPT
+# can be overwritten by shell variables, e.g. "BASE_SPARK_VERSION=3.1.2 MVN_OPT=bash build.sh".
+# Current accepted spark versions: [3.1.2, 3.2.1, 3.3.0].
+#
+# To add support of new runtime:
+#   1. Review `set_jars_prefixes()` to make sure that the prefix of the jar files is set
+#      correctly. If not, then add a new if-else block to set the variables as necessary.
+#   2. Review `set_sw_versions(). If the new runtime depends on different versions, then add a new
+#      block to define the correct versions.
+#   3. The jar files and their artifacts are defined in `set_dep_jars()`.
+#      You may need to add another conditional block because some runtimes may require special
+#      handling.
+#      For example, "3.1.2" had different patters for a few JARs (i.e., HIVE).
+#   4. If you had to go beyond the above steps to support the new runtime, then update the
+#      instructions accordingly.
 
-$MVN_CMD -B -Ddatabricks -Dbuildver=$BUILDVER clean package -DskipTests $MVN_OPT
+set -ex
 
+# Default SPARK Version is extracted from DB Spark environment.
+DEFAULT_SPARK_BASE_VERSION=$(< /databricks/spark/VERSION)
```

Comment:
Let us merge the HEAD of branch-22.12 into the PR branch

Reply:
I merged it. Do you want to get rid of the `DEFAULT_SPARK_BASE_VERSION` ?

Reply:
sorry missed that you just refactored on top of the upmerge. the change looks fine.